.github/workflows/Regression.yml

```diff
@@ -65,11 +65,34 @@ jobs:
           key: ${{ github.job }}
           save: ${{ github.ref == 'refs/heads/main' || github.repository != 'duckdb/duckdb' }}
 
-      - name: Build
+      - name: Checkout Private Regression
+        if: ${{ github.repository == 'duckdb/duckdb' && github.ref == 'refs/heads/main' }}
+        uses: actions/checkout@v4
+        with:
+          repository: duckdblabs/fivetran_regression
+          ref: main
+          token: ${{ secrets.DUCKDBLABS_BOT_TOKEN }}
+          path: benchmark/fivetran
+
+      # For PRs we compare against the base branch
+      - name: Build Current and Base Branch
+        if: ${{ !(github.repository == 'duckdb/duckdb' && github.ref == 'refs/heads/main') }}
         shell: bash
         run: |
-          make
           git clone --branch ${{ env.BASE_BRANCH }} https://github.com/duckdb/duckdb.git --depth=1
+          make
+          cd duckdb
+          make
+          cd ..
+
+      # For NightlyTest we fetch the last commit hash that successfully completed Regression on main
+      - name: Build Main and Previous Successful Regression Hash
+        if: ${{ github.repository == 'duckdb/duckdb' && github.ref == 'refs/heads/main' }}
+        shell: bash
+        run: |
+          git clone https://github.com/duckdb/duckdb.git
+          git checkout $(gh run list --repo duckdb/duckdb --branch=main --workflow=Regression --status=success --json=headSha --limit=1 --jq '.[0].headSha')
```

Comment:
I am not sure of the syntax here, I think it should be:
```
git clone https://github.com/duckdb/duckdb.git
cd duckdb
git checkout $(gh run list --repo duckdb/duckdb --branch=main --workflow=Regression --status=success --json=headSha --limit=1 --jq '.[0].headSha')
cd ..
make
cd duckdb
make
cd ..
```
this is since you need to checkout the repo in the duckdb folder / keep the workflows as similar as possible (same direction of what's new and what's old)